src/components/LibraryItem.tsx

```diff
@@ -54,7 +57,21 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
 
     constructor(props: LibraryItemProps) {
         super(props);
-        this.state = { expanded: false }; // Assign initial state
+
+        // Assign initial state
+        this.state = {
+            expanded: this.props.data.expanded,
+            visible: this.props.data.visible
+        };
+    }
+
+    componentWillReceiveProps(nextProps: LibraryItemProps) {
+        if (nextProps.data.visible !== this.state.visible) {
+            this.setState({ visible: nextProps.data.visible });
+        }
+        if (nextProps.data.expanded !== this.state.expanded) {
+            this.setState({ expanded: nextProps.data.expanded });
+        }
```

Comment:
We will need this if we want to allow user to toggle the expansion in the structured search view. Since by default all the found items in search view have `expanded = true` stored in their `ItemData`, we also have a `expanded` state in the state of `LibraryItem`, this is needed to make sure that these two always the same. Otherwise there will be issues like which one should we check when rendering. However, if we just don't allow user to unexpand in search view, this will not be needed.